### ISSUE TITLE
Battery sensors became binary sensors 

### ISSUE THREAD

**Issue by therealhalifax** [tschamm/boschshc-hass#30]
With the last update the Battery sensors became binary sensors, which is a bit confusing. I read the other issue from Februar where you explained that the sensor reports only normal, low and critical low. But now it is reduced just to 2 states. Which state is used for fault state? Low or critical low?

**Comment by t3t3** [comment]
keen to know that too; really hope the low (20%) state is covered since this is imo most important (in order to replace batteries in time, not if they have died already).

personally i think i.e. a gauge showing 100/20/0 is still better than an on/off icon (whatever it might mean in particular)...

**Comment by tschamm** [comment]
I agree on that, however the binary behavior was required by ha-core dev team.
The battery state is reported `off` as soon as the reported battery state is `not ok`. So it covers all three states 20/0/not_available. For the states 20/0, there will be reported a warning.

If you have suggestions how to further improve this, I would be happy to read.

**Comment by Marow** [comment]
Is it possible to show percentage of the battery instead of like "normal" or so?

**Comment by tschamm** [comment]
It is not, as the SHC is only reporting battery level as an enum.

**Comment by mosandlt** [comment]
Addressed in **0.5.0**: a granular **battery-level diagnostic sensor** (OK / LOW_BATTERY / CRITICAL_LOW / CRITICALLY_LOW_BATTERY / NOT_AVAILABLE) is now created alongside the existing low-battery binary sensor, restoring the severity detail that was lost when battery became binary-only. Update via HACS. (Diagnostic sensors can be toggled under the integration's options.)

**Comment by mosandlt** [comment]
Fixed in **boschshc-hass 0.5.0** — battery level is exposed again (as a diagnostic sensor).

Please update (via HACS, or manually — see #279) and restart Home Assistant. If you're on ≥0.5.0 and still hit this, comment here and we'll dig back in; otherwise I'll close this in ~2 weeks. Thanks!

**Comment by mosandlt** [comment]
Battery level is back as a proper diagnostic `sensor` (≥0.5.0), not a binary sensor.

Available in current releases — please update to **boschshc-hass 0.7.6**. Closing; reopen if anything is missing.
